Pyrado/scripts/training/qq-su_pddr.py

```diff
@@ -0,0 +1,219 @@
+# OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+# IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+"""
+Train an agent to solve the Quanser Qube swing-up task using Policy Distillation with Domain Randomization.
+"""
+import torch as to
+from torch.optim import lr_scheduler
+
+import pyrado
+from pyrado.algorithms.step_based.gae import GAE
+from pyrado.spaces import ValueFunctionSpace
+from pyrado.algorithms.meta.pddr import PDDR, PDDRTeachers
+from pyrado.algorithms.step_based.ppo import PPO
+from pyrado.environment_wrappers.action_normalization import ActNormWrapper
+from pyrado.environments.pysim.quanser_qube import QQubeSwingUpSim
+from pyrado.logger.experiment import setup_experiment, save_dicts_to_yaml
+from pyrado.policies.feed_forward.fnn import FNNPolicy
+from pyrado.utils.data_types import EnvSpec
+
+import argparse
+import multiprocessing as mp
+from multiprocessing.spawn import freeze_support
+
+from pyrado.domain_randomization.default_randomizers import create_default_randomizer
+from pyrado.environment_wrappers.domain_randomization import DomainRandWrapperLive
+from pyrado.policies.special.environment_specific import QCartPoleSwingUpAndBalanceCtrl, QQubeSwingUpAndBalanceCtrl
+from pyrado.environments.pysim.quanser_ball_balancer import QBallBalancerSim
+from pyrado.environments.pysim.quanser_cartpole import QCartPoleSwingUpSim
+
+if __name__ == "__main__":
+    # For multiprocessing and float32 support, recommended to include at top of script
+    freeze_support()
+    to.set_default_dtype(to.float32)
+
+    parser = argparse.ArgumentParser()
+
+    # Environment
+    parser.add_argument("--frequency", type=int, default=250)
+    parser.add_argument("--max_iter_teacher", type=int, default=200)
+    parser.add_argument("--max_iter", type=int, default=500)
+    parser.add_argument("--env_name", type=str, default="qq-su")
+    parser.add_argument("--num_teachers", type=int, default=2)
+    parser.add_argument("--max_steps", type=int, default=1500)
+    parser.add_argument("--seed", type=int, default=None)
+    parser.add_argument("--train_teachers", action="store_true", default=False)
+    parser.add_argument("--eval_reps", type=int, default=0)
+    parser.add_argument("--device", type=str, default="cpu")
+
+    parser.add_argument("--num_epochs", type=int, default=10)
+
+    # Parse command line arguments
+    args = parser.parse_args()
+
+    # Set seed if desired
+    pyrado.set_seed(args.seed, verbose=True)
+    use_cuda = args.device == "cuda"
+    descr = f"_{args.max_steps}st_{args.frequency}Hz"
+
+    if args.train_teachers:
+
+        # Parameters
+
+        # Environment / Experiment (set seed before creating the modules) / Randomizer
+        env_hparams = dict(dt=1 / args.frequency, max_steps=args.max_steps)
+        if args.env_name == "qcp-su":
```

Comment:
This file is specific to the qq-su env. Please do not povide the option for different envs here. Create a serparate file for the qcp-su env (watch the top level comment and imports)